.github/workflows/create-release-branch.yml

```diff
@@ -66,27 +68,69 @@ jobs:
           else
             RELEASE_TYPE=minor
           fi
-          # Increment the version as needed
+          # Increment the version as needed.
           npm --no-git-tag-version version $RELEASE_TYPE
-          git add package.json
-      # Update the dependencies
-      - name: Update dependencies
-        run: |
-          yarn upgrade
-          node tools/UpdateRetireJsVulns.js
-          git add yarn.lock
-          git add retire-js
-      # Create the new branch
-      - name: Checkout new branch, commit, push
-        id: create-branch
+      # Create the new branch as a direct copy of `dev` and push it to GitHub so the API can
+      # interact with it later.
+      - id: create-branch
         run: |
           NEW_VERSION=$(jq -r ".version" package.json)
+          git stash
+          git checkout dev
+          git pull
           git checkout -b release-$NEW_VERSION
-          git config --global user.name "sfca-bot"
-          git config --global user.email "cli-scanner@salesforce.com"
-          git commit -m "Incrementing version for $NEW_VERSION release"
           git push --set-upstream origin release-$NEW_VERSION
+          git stash pop
+          # Output the branch name so we can use it in later jobs.
           echo "branch_name=release-$NEW_VERSION" >> "$GITHUB_OUTPUT"
+      # Update our dependencies.
+      - run: |
+          yarn upgrade
+          node tools/UpdateRetireJsVulns.js
+      # Use the GraphQL API to create a signed commit with the various changes.
+      - run: |
+          # GraphQL needs to know what branch to push to.
+          BRANCH_NAME=$(git rev-parse --abbrev-ref HEAD)
+          # GraphQL needs a message for the commit.
+          NEW_VERSION=$(jq -r ".version" package.json)
+          MESSAGE="Preparing for v$NEW_VERSION release."
+          # GraphQL needs the latest versions of the files we changed, as Base64 encoded strings.
+          NEW_PACKAGE="$(cat package.json | base64)"
+          NEW_YARN_LOCK="$(cat yarn.lock | base64)"
+          NEW_RETIREJS_VULNS="$(cat retire-js/RetireJsVulns.json | base64)"
+          gh api graphql -F message="$MESSAGE" -F oldOid=`git rev-parse HEAD` -F branch="$BRANCH" \
```

Comment:
Is there a resource where I can learn about this. Did the other methods for signing commits fail, thus leaving you with using the api directly?

Reply:
GraphQL overview: https://docs.github.com/en/graphql
GH API overview: https://cli.github.com/manual/gh_api

Reply:
Also, the reason I'm using this tactic is because the commits created by the GraphQL API are signed by default, with no need for us to store any GPG keys as repo-level secrets or create a bot user. It all just works.

Reply:
Great. Thank you.